### PR TITLE
GEN-204 Create js context

### PR DESCRIPTION
**Pull request by arman-mahtabfar**
## Change description

> Added endpoints for js context package

TO DO: Add CircleCi build?

## Checklist

- [x] Application changes have been tested and automated tests have been added, if applicable
- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

## Related issues

> Fixes #xxxx

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -3,9 +3,8 @@ services:
   app:
     build:
       context: .
-    working_dir: /opt/project
     volumes:
-      - .:/opt/project
+      - ./:/usr/local/src/project
```
Comment (meierj-wayscript) [inline docker-compose.yml:7]: Is there a reason you changed this? I'd rather have it point to the local copy. Maybe we can do a quick pairing session.

File: test/context.test.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    process.env.PROCESS_UUID = "lsdkfj";
+    expect(context.getProcessDetailExpandedData()).toStrictEqual(JSON.parse(`{ "Success" : "False", "Status" : "404"}`));
+});
+
+
+// To Do: Add more tests once there are valid endpoints
+
+
+/**
+test('Get Process', () => {
```
Comment (meierj-wayscript) [inline test/context.test.js]: I'm not sure how Jeff feels about this, but I generally prefer not to have commented-out code in master. In my mind master should be a pristine (or as pristine as possible) working copy of the codebase.

Could you put these tests in another branch and open a separate PR? Then we can merge them once the other endpoints are up.

**Review by meierj-wayscript — CHANGES_REQUESTED** [review]
Generally looks good! Let's sync up on the dockerfile changes.

File: src/setup.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+const WAYSCRIPT_ORIGIN = process.env.WAYSCRIPT_ORIGIN || "https://api.wayscript.com";
+
+const PROCESSES_ROUTE = "processes";
+const LAIRS_ROUTE = "lairs";
+const WORKSPACES_ROUTE = "workspaces";
+
+const ROUTES = {
+    "lairs": {"detail": LAIRS_ROUTE + "/$id"},
```
Comment (jdhayford) [inline src/setup.js]: Strings not necessary around keys in js object literal

File: src/context.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+function getProcessDetailExpandedData() {
+    let processUUID = utils.getProcessUUID();
+    let client = new utils.WayScriptClient();
+    
+    let response = client.getProcessDetailExpanded(processUUID);
+    return processResponseFromClient(response);
+}
+
+function getProcess() {
+    return getProcessDetailExpandedData()["process"];
```
Comment (jdhayford) [inline src/context.js]: Can access these via dot operators, `.process`

File: src/context.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+function getEvent() {
+    return getProcessDetailExpandedData()["event"];
```
Comment (jdhayford) [inline src/context.js]: Lets try to use single quotes for strings

File: test/context.test.js (hunk 2 above)
Comment (arman-mahtabfar) [reply test/context.test.js]: Made a separate branch for endpoint tests once they're up

File: docker-compose.yml (hunk 1 above)
Comment (arman-mahtabfar) [reply docker-compose.yml:7]: Removed docker changes

**Review by meierj-wayscript — APPROVED** [review]
LTGM 👍

**Pull request merged**